Formula/apache-pulsar.rb

```diff
@@ -0,0 +1,77 @@
+class ApachePulsar < Formula
+  desc "Cloud-native distributed messaging and streaming platform"
+  homepage "https://pulsar.apache.org/"
+  url "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.8.0/apache-pulsar-2.8.0-src.tar.gz"
+  mirror "https://archive.apache.org/dist/pulsar/pulsar-2.8.0/apache-pulsar-2.8.0-src.tar.gz"
+  sha256 "0e161a81c62c7234c1e0c243bb6fe30046ec1cd01472618573ecdc2a73b1163b"
+  license "Apache-2.0"
+  head "https://github.com/apache/pulsar.git", branch: "master"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cppunit" => :build
+  depends_on "libtool" => :build
+  depends_on "maven" => :build
+  depends_on "pkg-config" => :build
+  depends_on "protobuf" => :build
+  depends_on arch: :x86_64
+  depends_on "openjdk@11"
+
+  def install
+    chmod "+x", "src/rename-netty-native-libs.sh"
+    with_env(
+      "JAVA_HOME" => Formula["openjdk@11"].opt_prefix,
```

Comment:
This isn't the usual location for `JAVA_HOME` so I don't think this does anything.

Reply:
👍, addressed!

Reply:
How exactly? The line is still the same.

Reply:
Unfortunately removing this caused tests to fail; I restored it in the latest commit.

Reply:
Failures in https://github.com/Homebrew/homebrew-core/pull/83238/commits/1e325fb10c37fbcdd120a93cee45d338bcd15b61 went away when I put this line back in https://github.com/Homebrew/homebrew-core/pull/83238/commits/0d2389dca19a02fe8fb7d0306bf5bb13583c7d25

Reply:
Wait, the test block was failing? That means that it's compiling in this value. In that case especially this shouldn't be `opt_prefix` but `overridable_java_home`

Reply:
No, the install step was failing because it was building on a newer, incompatible version of Java. Evidence from the CI failure in the first linked commit:

```
==> mvn -X clean package -DskipTests -Pcore-modules
Picked up _JAVA_OPTIONS: -Duser.home=/Users/brew/Library/Caches/Homebrew/java_cache
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /usr/local/Cellar/maven/3.8.2/libexec
Java version: 16.0.2, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk/16.0.2/libexec/openjdk.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "11.3.1", arch: "x86_64", family: "mac"
[DEBUG] Created new class realm maven.api
```

Reply:
Apologies for the confusion; by "tests" I meant "formula CI in general", not `brew test`.